### PR TITLE
feat: initial client implementation

### PR DESCRIPTION
**Pull request by andreabadesso**
## Motivation

While integrating the mobile-wallet with [SES](https://github.com/endojs/endo/tree/master/packages/ses), I've detected that the `unleash-proxy-client-js` library attempts to mutate the javascript prototype, which is forbidden by default by the sandbox

We could whitelist it, but since the lib has many features that we don't use, I thought it would be a better idea to create our own, with a limited subset of features that we actually use in our services and with no dependencies

### Acceptance Criteria
- We should have a minimal version of an unleash client with only the methods we use
- We should be able to fetch the feature toggles
- We should be able to check wether a feature toggle is enabled or disabled for a given user
- We should use E-TAG to reduce bandwidth (by only downloading toggles if they changed)

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 node_modules/
 build/
 .direnv/
+test.js
```
Comment (pedroferreira1) [inline .gitignore]: Why did you add this?

File: src/utils/index.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * @param context - An object representing the context, 
+ *                             which contains key-value pairs to be added as query parameters.
+ *
+ * @returns A new URL object with appended query parameters based on the provided context.
+ *
+ * Example:
+ *   Given a URL 'http://example.com' and a context object 
+ *   { appName: 'wallet-mobile', clientKey: 'key', properties: { x: 'foo', y: 'bar' } },
+ *   this function will return a new URL: 
+ *   'http://example.com?appName=wallet-mobile&clientKey=&properties[x]=foo&properties[y]=bar'.
```
Comment (pedroferreira1) [inline src/utils/index.ts]:
```suggestion
 *   'http://example.com?appName=wallet-mobile&clientKey=key&properties[x]=foo&properties[y]=bar'.
```

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+   *
+   * If the toggles with the given name is not found, it returns `false`.
+   *
+   * An error is thrown if the toggles have not been loaded.
+   *
+   * @param name - The name of the toggle to check.
+   * @returns `true` if the toggle is found and enabled, otherwise `false`.
+   * @throws {Error} If the toggles have not been downloaded.
+   */
+  public isEnabled(name: string): boolean {
```
Comment (pedroferreira1) [inline src/index.ts:128]: Does it make sense to add here a parameter with a default value in case the toggle is not found? Or do you prefer to keep the same as the library?

How are we handling default values, in case we prefer that it's enabled by default?

File: .gitignore (hunk 1 above)
Comment (andreabadesso) [reply .gitignore]: This shouldn't have been commited, removed in 0f228c6f3b3eeaa14f96a1f6c357c875ad6691af


thanks

File: src/index.ts (hunk 3 above)
Comment (andreabadesso) [reply src/index.ts:128]: If the toggle is deactivated for this specific user, it will not be found here, unleash never sends a toggle with `enabled: false`, so we can't add a default for the case that the toggle is not found

We could have a default for when the toggles are not downloaded yet, but I prefer to throw an error

**Review by pedroferreira1 — APPROVED** [review]

**Review by alexruzenhack — APPROVED** [review]
